### PR TITLE
Filter column display width in schema cache dump

### PR DESCRIPTION
**Pull request by wjordan**
### Description
MySQL supports an [extension](https://dev.mysql.com/doc/refman/en/numeric-type-attributes.html) for optionally specifying the display width of integer data types in parentheses.

These attributes are deprecated in MySQL 8.0.17 and removed from the `show create table` output in 8.0.19, and so they will cause meaningless schema-cache diffs for anyone running newer versions of MySQL in their local development environment.

This PR filters these display-width attributes from the SQL type metadata that gets serialized to the schema-cache dump, so these diffs will no longer continue to occur.

### Implementation note
Note that this code change extends the existing patch created in #36151. This filter will probably be helpful to submit upstream to Rails eventually, but this local patch should work for now for our purposes.

### CODE REVIEW

File: dashboard/config/initializers/schema_cache_dedup.rb [hunk 1]
```diff
@@ -138,6 +138,9 @@ def initialize(type_metadata, extra: nil, strict: false)
     private
 
     def deduplicated
+      # MySQL supports an extension for optionally specifying the display width of integer data types in parentheses.
+      # Filter display width attributes from schema-cache, they are deprecated in MySQL 8.0.17 and cause meaningless diffs.
+      @sql_type.gsub!(/\(\d+\)/, '') if @type == :integer
```
Comment (jamescodeorg) [inline dashboard/config/initializers/schema_cache_dedup.rb]: Does type :integer also cover bigints?

**Review by jamescodeorg — APPROVED** [review]
LGTM, assuming that integer covers bigint.

Should we check-in an updated schema_cache.dump as part of this change?

**Comment by wjordan** [comment]
Added `schema_cache.dump` (2d185b4adb6001b680b48b9e19e4874b10a8dca9), and also a commit (0ca0d21ef332fcbeb9318b323634a293b294aa1a) containing some integration scripts to read the schema-cache dump file through `git diff` (and to hook it up through when running the existing `bundle exec rake install:hooks` task).

Here's the schema-cache diff from 2d185b4adb6001b680b48b9e19e4874b10a8dca9, noting that each `sql_type` string is only changed once in the diff because the underlying objects are deduplicated in the serialized object:

```patch
code-dot-org/dashboard/db$ git diff 2d185b4adb6001b680b48b9e19e4874b10a8dca9~1 -- schema_cache.dump
diff --git a/dashboard/db/schema_cache.dump b/dashboard/db/schema_cache.dump
index ed9e83b14bc..04fa5af36c3 100644
--- a/dashboard/db/schema_cache.dump
+++ b/dashboard/db/schema_cache.dump
@@ -7,7 +7,7 @@ columns:
     table_name: activities
     sql_type_metadata: &7 !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
       type_metadata: &1 !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: int(11)
+        sql_type: int
         type: :integer
         limit: 4
         precision: 
@@ -24,12 +24,7 @@ columns:
     name: user_id
     table_name: activities
     sql_type_metadata: &2 !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
-      type_metadata: !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: int(11)
-        type: :integer
-        limit: 4
-        precision: 
-        scale: 
+      type_metadata: *1
       extra: ''
       strict: true
       delegate_dc_obj: *1
@@ -261,7 +256,7 @@ columns:
     table_name: ap_cs_offerings
     sql_type_metadata: &20 !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
       type_metadata: &10 !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: smallint(6)
+        sql_type: smallint
         type: :integer
         limit: 2
         precision: 
@@ -426,7 +421,7 @@ columns:
     table_name: assessment_activities
     sql_type_metadata: &15 !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
       type_metadata: &13 !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: bigint(11) unsigned
+        sql_type: bigint unsigned
         type: :integer
         limit: 8
         precision: 
@@ -5487,7 +5482,7 @@ columns:
     table_name: pd_misc_surveys
     sql_type_metadata: &38 !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
       type_metadata: &37 !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: bigint(20)
+        sql_type: bigint
         type: :integer
         limit: 8
         precision: 
@@ -10256,12 +10251,7 @@ columns:
     name: id
     table_name: user_levels
     sql_type_metadata: !ruby/object:ActiveRecord::ConnectionAdapters::MySQL::TypeMetadata
-      type_metadata: !ruby/object:ActiveRecord::ConnectionAdapters::SqlTypeMetadata
-        sql_type: bigint(11) unsigned
-        type: :integer
-        limit: 8
-        precision: 
-        scale: 
+      type_metadata: *13
       extra: auto_increment
       strict: true
       delegate_dc_obj: *13

```

File: dashboard/config/initializers/schema_cache_dedup.rb (hunk 1 above)
Comment (wjordan) [reply dashboard/config/initializers/schema_cache_dedup.rb]: Yes- looks like [`abstract_mysql_adapter`](https://github.com/rails/rails/blob/c4d3e202e10ae627b3b9c34498afb45450652421/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L671-L675) registers all the various MySQL numeric data-types as an ActiveRecord `:integer` type.

**Comment by jamescodeorg** [comment]
Very cool!  Thanks for adding the human-readable diff script!

**Pull request merged**